### DISCUSSION TITLE
Drop-down value is not selecting

### DISCUSSION THREAD

**Discussion by ViranshDubey** [cypress-io/cypress#26037] [Questions and Help]
Hello, 

I have an issue with the drop-down value selection. The drop-down values open but not selected even though the CLICK command is successful. Kindly help.

Please have a look at the attached screenshot.

`cy.get("sap-crm-drop-down-button[arialabel='Edit ']").parent("sap-crm-drop-down").click();
cy.contains("sap-crm-list-item", "completionDueOn").should("be.visible").click();`

Regards,
Vishal
<img width="1439" alt="Cypress-service-levels" src="https://user-images.githubusercontent.com/113082620/223361918-b889bd73-0b78-4629-a1d9-7831e57e78b9.png">
<img width="1516" alt="Cypress-service-levels2" src="https://user-images.githubusercontent.com/113082620/223363198-1f33d66b-8c58-40ab-be04-ee814d68b19b.png">

**Comment by portyu9** [comment]
Hi @ViranshDubey 

The screenshots show that `sap-crm-list-item` is a **custom element with a shadow root**.

So this:

```js
cy.contains('sap-crm-list-item', 'completionDueOn')
  .should('be.visible')
  .click()
```

is clicking the custom-element host. The actual selectable element shown in DevTools is the `<li role="listitem">` **inside its shadow DOM**.

Try clicking that element directly:

```js
cy.contains('sap-crm-list-item', 'completionDueOn')
  .shadow()
  .find('li[role="listitem"]')
  .should('be.visible')
  .click()
```

Then assert the state that should change after selection, for example the selected label/value.

I would not make `{ force: true }` the first solution here. Cypress is already reporting that the click occurred; the more important detail from the DOM screenshot is that the event needs to reach the actual interactive element inside the shadow root.

Please mark this answer as accepted if it helped, thank you.
